### PR TITLE
Js tasks

### PR DESCRIPTION
**Pull request by ZamaSharik**

### CODE REVIEW

File: HomeTask-2/JSTask-1/transportationOnVacation.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+function rentalCarCost(d) {
+  var everyDayCost = 40;
+  var discount;
+
+  var totalAmount = d * everyDayCost;
+
+  if(d >= 7) {
+    totalAmount = totalAmount - 50;
+    return totalAmount;
```
Comment (SaiTeja2904) [inline HomeTask-2/JSTask-1/transportationOnVacation.js]: Can remove multiple return statements and make it one.

File: HomeTask-2/JSTask-2/getTheMiddleCharacter.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+function getMiddle(s) {
+  //Code goes here!
+  var len = s.length;
+  var mid = len / 2;
+
+  if (len == 1) {
+    return s;
+  }
+
+  if (len % 2 == 0) {
```
Comment (SaiTeja2904) [inline HomeTask-2/JSTask-2/getTheMiddleCharacter.js]: Use === instead of ==

File: HomeTask-2/JSTask-2/getTheMiddleCharacter.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    return s.substring(Math.trunc(mid) - 1, Math.trunc(mid) + 1);
```
Comment (SaiTeja2904) [inline HomeTask-2/JSTask-2/getTheMiddleCharacter.js:15]: Extract similar lines of code to another function.

File: HomeTask-2/JSTask-2/partitionOn.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/*
+Problem link: http://www.codewars.com/kata/partition-on
+*/
+
+// partition the items array so that all values for which pred returns true are
+// at the end, returning the index of the first true value
+function partitionOn(pred, items) {
```
Comment (SaiTeja2904) [inline HomeTask-2/JSTask-2/partitionOn.js:7]: Explore the different array methods and try solving this in another approach.

File: HomeTask-2/JSTask-2/getTheMiddleCharacter.js (hunk 3 above)
Comment (ZamaSharik) [reply HomeTask-2/JSTask-2/getTheMiddleCharacter.js:15]: Resolved.

File: HomeTask-2/JSTask-2/partitionOn.js (hunk 4 above)
Comment (ZamaSharik) [reply HomeTask-2/JSTask-2/partitionOn.js:7]: resolved. Please have a look.

File: HomeTask-2/JSTask-1/transportationOnVacation.js (hunk 1 above)
Comment (ZamaSharik) [reply HomeTask-2/JSTask-1/transportationOnVacation.js]: Resolved.

File: HomeTask-2/JSTask-1/transportationOnVacation.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+*/
+
+function rentalCarCost(d) {
+  var everyDayCost = 40;
+  var discount;
+
+  var totalAmount = d * everyDayCost;
+
+  if(d >= 7) {
+    totalAmount = totalAmount - 50;
```
Comment (SaiTeja2904) [inline HomeTask-2/JSTask-1/transportationOnVacation.js]: Can use short hand notation. a-=50;

File: HomeTask-2/JSTask-2/partitionOn.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  var evenItemsCopy = items.filter(pred);
+  var oddItemsCopy = items.filter(
+    function(value) {
+      return !pred(value);
```
Comment (SaiTeja2904) [inline HomeTask-2/JSTask-2/partitionOn.js]: this can be written as items.filter(val => !pred(val))

**Review by SaiTeja2904 — APPROVED** [review]

File: HomeTask-2/JSTask-2/partitionOn.js (hunk 6 above)
Comment (ZamaSharik) [reply HomeTask-2/JSTask-2/partitionOn.js]: That looks good! I'll go through arrow functions documentation and will do the changes. Thank you.

File: HomeTask-2/JSTask-1/transportationOnVacation.js (hunk 5 above)
Comment (ZamaSharik) [reply HomeTask-2/JSTask-1/transportationOnVacation.js]: Made the changes.

**Review by SaiTeja2904 — APPROVED** [review]

**Pull request merged**